Server/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -898,23 +898,33 @@ int CLuaResourceDefs::getResourceLoadTime(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::getResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    CResource* pResource;
+    if (!resourceElement)
+    {
+        CLuaMain*  localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+        CResource* localResource;
```

Comment:
It's C++, not C89, there's no need in variables declaration before being use. You can declare this varible and initialize it in one shot at line 911.

Reply:
This is my preferred way, but it doesn't matter to me.